File: ueueo-framework/ueueo-background-workers/src/main/java/com/ueueo/BackgroundWorkers/BackgroundWorkerManager.cs
using System;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;
using Volo.Abp.DependencyInjection;
using Volo.Abp.Threading;

namespace Volo.Abp.BackgroundWorkers;

/**
 * Implements <see cref="IBackgroundWorkerManager"/>.
*/
public class BackgroundWorkerManager : IBackgroundWorkerManager, ISingletonDependency, IDisposable
{
    protected boolean IsRunning ;// { get; private set; }

    private boolean _isDisposed;

    private readonly List<IBackgroundWorker> _backgroundWorkers;

    /**
     * Initializes a new instance of the <see cref="BackgroundWorkerManager"/> class.
    */
    public BackgroundWorkerManager()
    {
        _backgroundWorkers = new List<IBackgroundWorker>();
    }

    public void AddAsync(IBackgroundWorker worker)
    {
        _backgroundWorkers.Add(worker);

        if (IsRunning)
        {
            worker.StartAsync();
        }
    }

    public void Dispose()
    {
        if (_isDisposed)
        {
            return;
        }

        _isDisposed = true;

        //TODO: ???
    }

    public void StartAsync(CancellationToken cancellationToken = default)
    {
        IsRunning = true;

        for (var worker in _backgroundWorkers)
        {
            worker.StartAsync(cancellationToken);
        }
    }

    public void StopAsync(CancellationToken cancellationToken = default)
    {
        IsRunning = false;

        for (var worker in _backgroundWorkers)
        {
            worker.StopAsync(cancellationToken);
        }
    }
}

File: ueueo-framework/ueueo-background-workers/src/main/java/com/ueueo/backgroundworkers/BackgroundWorkerManager.java
package com.ueueo.backgroundworkers;

import com.ueueo.IDisposable;

import java.util.ArrayList;
import java.util.List;

/**
 * Implements <see cref="IBackgroundWorkerManager"/>.
 */
public class BackgroundWorkerManager implements IBackgroundWorkerManager, IDisposable {
    protected boolean isRunning;

    private boolean isDisposed;

    private List<IBackgroundWorker> backgroundWorkers;

    /**
     * Initializes a new instance of the <see cref="BackgroundWorkerManager"/> class.
     */
    public BackgroundWorkerManager() {
        backgroundWorkers = new ArrayList<>();
    }

    @Override
    public void add(IBackgroundWorker worker) {
        backgroundWorkers.add(worker);

        if (isRunning) {
            worker.start();
        }
    }

    @Override
    public void dispose() {
        if (isDisposed) {
            return;
        }

        isDisposed = true;

        //TODO: ???
    }

    @Override
    public void start() {
        isRunning = true;

        for (IBackgroundWorker worker : backgroundWorkers) {
            worker.start();
        }
    }

    @Override
    public void stop() {
        isRunning = false;

        for (IBackgroundWorker worker : backgroundWorkers) {
            worker.stop();
        }
    }

    public boolean isRunning() {
        return isRunning;
    }
}
